FreeRTOS/FreeRTOS-Plus-TCP
# IPV6/multi: Reduced complexity FreeRTOS_DHCPv6

Description
-----------
This PR want to reduce the complexity of FreeRTOS_DHCPv6.c.

To following 3 functions now call a helper function:

vDHCPv6Process()  score 29
- xDHCPv6Process_PassReplyToEndPoint()

vDHCPv6ProcessEndPoint()  score 62
- xDHCPv6ProcessEndPoint_HandleState()
	- vDHCPv6ProcessEndPoint_HandleReply()
	- xDHCPv6ProcessEndPoint_HandleAdvertise()
	- vDHCPv6ProcessEndPoint_SendDiscover()

prvDHCPv6Analyse()  score 191
- prvDHCPv6_handleOption()
	- prvDHCPv6_subOption()

So in two cases, the helper functions call another helper function.

Test Steps
-----------
I tested the changes by having devices starting up with DHCPv6 enabled and analyse the communication with WireShark.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
